### DISCUSSION TITLE
[SCENE] Tableau de bord

### DISCUSSION THREAD

**Discussion by WITNicolas** [witsa/synapps#561] [Show and tell]
L'acteur tableau de bord n'existant pas encore, voici une scene qui permet d'afficher cela.
![image](https://user-images.githubusercontent.com/33632664/212627986-be6c36f0-88e3-401f-9486-48f9c4be2de1.png)

~~~~text
SYNAPPS-STUDIO-SCENE|{"config":{"key":"dashboard","name":"Tableau de bord"},"leadActor":{"type":"layout/stack","key":"stack1","children":[{"type":"layout/stack","key":"stack2","children":[{"type":"display/iframe","key":"iframe1","properties":{"verticalAlignment":"expand"},"additionalDefs":{"redyPath":{"type":"text","label":"Chemin REDY"}},"additionals":{"redyPath":"/easy/dashboard"},"bindings":{"properties.url":"relative#self@additionals.redyPath"},"events":{"properties/url/binding/onReadTransform":["return  `${this.dataStores.redy.host.domain}/WSID${this.dataStores.redy.host.session.sid}${context.value}`;"]}}],"properties":{"verticalAlignment":"expand"}}],"bindings":{"properties.backgroundImage":"color@gradient"}}}

**Comment by edejean** [comment]
Bonjour,
Possibilité de garder cette scène unique et d'appeler celle-ci de manière dynamique en précisant le tableau de bord à afficher depuis d'autres scène ?
Ce type de chemin par exemple : "/easy/Dashboard?NodId=261767&"
J'ai créé une additionnelle au niveau de la scène "Tableau de bord" mais je ne connais pas l'équivalent du getActor, j'ai essayé 👍 context.getScene('dashboard').additionals.numtdb= "/easy/Dashboard?NodId=261767&";
mais j'ai quelques doutes sur le getScene.... et surtout l'associer à quelle type d'évènement ? OnClick ?

**Comment by WITNicolas** [comment]
Bonjour,
Il n'y a pas besoin de script pour passer un argument.
Il suffit d'afficher la Scene Dashboard dans un acteur Ecran.
L'Ecran hérite naturellement de l'additionelle de la Scene Dashboard.
Et l'appel de la scene dashboard (dans l'Ecran) à partir d'un acteur de type Bouton de Navigation hérite lui aussi de l'additionnelle.

On peut donc préciser pour chaque bouton les arguments à passer qui nous interesse.

**Reply by edejean** [reply]
Par contre, si je ne veux pas afficher cet Ecran mais le déclencher au clic, il faut que je le mette dans une modale associé à un évènement OnClick ?

**Reply by WITNicolas** [reply]
Pourquoi pas effectivement c'est une solution :)
Une modale au click associeée au bouton de navigation et le bouton de navigation qui appelle la scene Dashaboard pour l'afficher dans l'acteur Ecran qui sera dans la modale.
Les arguments vont transiter depuis le bouton, vers l'ecran qui va passer à la scene ....
